packages/grid/src/GridMetricCalculator.ts

```diff
@@ -1745,13 +1745,20 @@ export class GridMetricCalculator {
       row => {
         const modelRow = this.getModelRow(row, state);
         const text = model.textForCell(modelColumn, modelRow);
+        const cellRenderType = model.renderTypeForCell(modelColumn, modelRow);
+
         if (text) {
           const cellPadding = cellHorizontalPadding * 2;
           columnWidth = Math.max(
             columnWidth,
             text.length * fontWidth + cellPadding
           );
         }
+
+        if (cellRenderType === 'dataBar') {
+          // columnWidth = Math.max(columnWidth, columnWidth + 90);
+          hasDataBar = true;
+        }
```

Comment:
Ah yea, I'd rather have the logic contained in here, rather than setting `hasDataBar` and adding it at the end. Reason being that as we add different cell renderers, the logic will apply on a cell by cell basis, and we may want to (/should) delegate the metric calculation to the renderer (or a `CellMetricCalculator` registered for that `renderType`). We should also define `dataBarHorizontalPadding` within the theme.
For now we can just do something like:
```
        const cellRenderType = model.renderTypeForCell(modelColumn, modelRow);

        let cellWidth = 0;
        if (text) {
          const cellPadding = cellHorizontalPadding * 2;
          cellWidth = text.length * fontWidth + cellPadding;
        }

        if (cellRenderType === 'dataBar') {
          cellWidth += dataBarHorizontalPadding;
        }
        columnWidth = Math.max(columnWidth, cellWidth);    
```